app/routes/city-council-districts.$cityCouncilDistrictId.capital-projects.tsx

```diff
@@ -0,0 +1,78 @@
+import { LoaderFunctionArgs, json } from "@remix-run/node";
+import { findCapitalProjectsByCityCouncilId, findAgencies } from "../gen";
+import { useLoaderData, useNavigate } from "@remix-run/react";
+import { CapitalProjectsAccordionPanel } from "../components/CapitalProjectsList";
+import { Box, Hide, Show, Text } from "@nycplanning/streetscape";
+import { CapitalProjectsDrawer } from "~/components/CapitalProjectsList/CapitalProjectsDrawer";
+
+export async function loader({ request, params }: LoaderFunctionArgs) {
+  const url = new URL(request.url);
+  const path = url.pathname;
+  const agenciesResponse = await findAgencies({
+    baseURL: `${import.meta.env.VITE_ZONING_API_URL}/api`,
+  });
+  const limitParam = url.searchParams.get("limit");
+  const offsetParam = url.searchParams.get("offset");
+  let limit;
+  let offset;
+  if (limitParam === null) {
+    limit = 20;
+  } else {
+    limit = parseInt(limitParam);
+  }
+  if (offsetParam === null) {
+    offset = 0;
+  } else {
+    offset = parseInt(offsetParam);
+  }
+
+  const { cityCouncilDistrictId } = params;
+  if (cityCouncilDistrictId === undefined) {
+    throw json("Bad Request", { status: 400 });
+  }
+
+  const projectsByCityCouncilDistrictResponse =
+    await findCapitalProjectsByCityCouncilId(
+      cityCouncilDistrictId,
+      {
+        limit: 7,
+        offset,
+      },
+      {
+        baseURL: `${import.meta.env.VITE_ZONING_API_URL}/api`,
+      },
+    );
+
+  return {projectsByCityCouncilDistrictResponse, agencies: agenciesResponse.agencies, limit, offset, path, cityCouncilDistrictId};
```

Comment:
I suggest focusing the return values of the loader to server-specific data. This would translate to removing the "limit", "offset", "path", and "cityCouncilDistrictId" from here. If we need them again in the component, we can still access them through the component-specific hooks like "useSearchParams". 

The immediate practical benefit would be to clarify which values come from where. The API response is paginated and will have its own "limit" and "offset" parameters. Removing the client's "limit" and "offset" from the loader response may help clarify that they're coming from the client url and not the API response.